src/NServiceBus.Core/Routing/Routers/UnicastSendRouterConnector.cs

```diff
@@ -34,7 +34,7 @@ class UnicastSendRouterConnector : StageConnector<OutgoingSendContext, OutgoingL
             var destination = state.ExplicitDestination ?? (state.RouteToLocalInstance ? localAddress : null);
 
             var addressLabels = string.IsNullOrEmpty(destination) 
-                ? unicastRouter.Route(messageType, distributionStrategy, context) 
+                ? await unicastRouter.Route(messageType, distributionStrategy, context) 
```

Comment:
I think missing `ConfigureAwait(false)` but not sure because I'm compiling in my head.